### ISSUE TITLE
End of line errors

### ISSUE THREAD

**Issue by GaryJones** [stylelint/stylelint#363]
![screenshot 2015-09-03 17 46 46](https://cloud.githubusercontent.com/assets/88371/9664746/c43df3d6-5263-11e5-9797-761aec54c1c3.png)

stylelint 0.8.0
stylelint-config-wordpress 0.1.0

> Expected newline after ";" (declaration-block-semicolon-newline-after)
> Expected newline after "{" (block-opening-brace-newline-after)
> Expected newline after "}" (block-closing-brace-newline-after)

I'm getting each of the above errors for pretty much each `{`, `;` and `}` in my file, which would suggest a discrepency in line-endings. My CSS file is generated as LF, but I've also tried CR and CRLF and still the errors keep coming. I use a fairly standard formatting:

```css
.member-login {
	position: relative;
	top: 18px;
}
```
The code already runs through [grunt-wp-css](https://github.com/cedaro/grunt-wp-css) (wrapper for csscomb), so is consistently formatted (i.e. the stylelint reported errors shouldn't exist!)

Any suggestions as to the cause of the reported errors?

**Comment by ntwb** [comment]
This looks like it's an issue with v0.6.0 of https://github.com/nDmitry/grunt-postcss

If I switch back to v0.5.5 everything is fine with the CSS example you provided.

Working as expected for me with:
```
    "grunt-postcss": "~0.5.5",
    "postcss-reporter": "~1.1.0",
    "stylelint": "~0.8.0",
    "stylelint-config-wordpress": "~0.2.0"
```

Fails when using::
```
    "grunt-postcss": "~0.6.0",
    "postcss-reporter": "~1.1.0",
    "stylelint": "~0.8.0",
    "stylelint-config-wordpress": "~0.2.0"
```

I also get a bunch of derecated errors when using v0.6.0 of grunt-postcss:
```
Container#eachComment is deprecated. Use Container#walkComments instead.
Container#eachAtRule is deprecated. Use Container#walkAtRules instead.
Container#eachRule is deprecated. Use Container#walkRules instead.
Node#before is deprecated. Use Node#raws.before
Node#between is deprecated. Use Node#raws.between
Node#after is deprecated. Use Node#raws.after
Container#eachDecl is deprecated. Use Container#walkDecls instead.
Node#semicolon is deprecated. Use Node#raws.semicolon
Container#eachInside is deprecated. Use Container#walk instead.
```

**Comment by ntwb** [comment]
This indeed looks like it's related to PostCSS v5.x and v4.x incompatibilities.

For now the best workaround appears to be to not mix PostCSS plugins that require v4 and v5, only use plugins that use v4 for now until all the plugins wanting to be used have been updated for PostCSS v5.

A rough list that should work well together for PostCSS v4

• stylelint v0.8.0 
• stylelint-config-wordpress v0.2.0
• grunt-postcss v0.5.5
• autoprefixer v5.2.1

**Comment by GaryJones** [comment]
Thanks for debugging. I'll bump down grunt-postcss and report back.

**Comment by GaryJones** [comment]
Switching to grunt-postcss 0.5.5. fixed it. Looks like 0.5.5 -> 0.6.0 had a [bump](https://github.com/nDmitry/grunt-postcss/commit/5f486be91386878702e8309f8f29b313ff13d955) in dependencies.

**Comment by jeddy3** [comment]
@ntwb Thanks for getting to the bottom of this one :)

@GaryJones FYI, work towards postcss 5.x compatibility is happening in issue #349 .

**Comment by jeddy3** [comment]
@GaryJones FYI, PostCSS `5.x` compatibility has just been released in version `1.0.0`.

**Comment by GaryJones** [comment]
Thanks for the heads-up.
